collector-exoplayer/src/main/java/com/bitmovin/analytics/exoplayer/ExoPlayerExceptionMapper.kt

```diff
@@ -5,65 +5,59 @@ import com.bitmovin.analytics.data.ErrorData
-                } else if (exception is BehindLiveWindowException) {
-                    errorCode = ErrorCode.BEHIND_LIVE_WINDOW
-                    errorCode.errorData = ErrorData("Behind live window: required segments not available")
-                }
 
-                if (errorCode.errorData == null) {
-                    errorCode.errorData = ErrorData(exception::class.java.toString() + ": " + (exception.message
-                            ?: ""), exception.topOfStacktrace)
+    private val errorMessages = mapOf(
+            -1 to "Unknown Error",
+            0 to "Source Error",
+            1 to "Render Error",
+            2 to "Unexpected Error",
+            3 to "Remote Error",
+            4 to "Out of memory Error")
+
+    override fun map(throwable: Throwable): ErrorCode {
+
+        val type = getExceptionType(throwable)
+        var message = errorMessages[type] ?: "Unknown Error"
+        var errorData: ErrorData? = null
+
+        val cause = throwable.cause
+        if (cause != null) {
+            when (cause) {
+                is HttpDataSource.InvalidResponseCodeException -> {
+                    message += ": InvalidResponseCodeException"
+                    errorData = ErrorData("Data Source request failed with HTTP status: " + cause.responseCode + " - " + cause.dataSpec.uri, cause.topOfStacktrace)
                 }
+                is HttpDataSource.InvalidContentTypeException -> {
+                    message += ": InvalidContentTypeException"
+                    errorData = ErrorData("Invalid Content Type: " + cause.contentType, cause.topOfStacktrace)
+                }
+                is HttpDataSource.HttpDataSourceException -> {
+                    message += ": HttpDataSourceException"
+                    errorData = ErrorData("Unable to connect: " + cause.dataSpec.uri, cause.topOfStacktrace)
+                }
+                is BehindLiveWindowException -> {
+                    message += ": BehindLiveWindowException"
+                    errorData = ErrorData("Behind live window: required segments not available", cause.topOfStacktrace)
+                }
+                else -> errorData = null
             }
-            TYPE_RENDERER -> {
-                errorCode = ErrorCode.EXOPLAYER_RENDERER_ERROR
-                errorCode.errorData = ErrorData(error.rendererException.message
-                        ?: "", error.rendererException.topOfStacktrace)
-            }
-            TYPE_UNEXPECTED -> {
-                errorCode = ErrorCode.UNKNOWN_ERROR
-                errorCode.errorData = ErrorData(error.unexpectedException.message
-                        ?: "", error.unexpectedException.topOfStacktrace)
-            }
-            else -> {
-                errorCode = ErrorCode.UNKNOWN_ERROR
-                errorCode.errorData = ErrorData(error.message
-                        ?: "", error.topOfStacktrace)
+            if (errorData == null) {
```

Comment:
move to `else` block

Reply:
https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/108/commits/7aaceb472e46119c739187f5e752b3a2da328991